src/graph.rs

```diff
@@ -2392,7 +2392,19 @@ fn analyze_dynamic_arg_template_parts(
           }
         }
         DirEntryKind::Dir => {
-          pending_dirs.push_back(entry.url);
+          // ignore hidden directories and any node_modules/vendor folders
+          let is_allowed_dir = entry
+            .url
+            .path()
+            .rsplit('/')
+            .find(|c| !c.is_empty())
+            .map(|c| {
+              !c.starts_with('.') && c != "node_modules" && c != "vendor"
```

Comment:
This somewhat aligns with what the CLI does here:

https://github.com/denoland/deno/blob/393abed3873d83019feb5bcebb10a6929133862a/cli/util/fs.rs#L322-L324

I wonder if we should match it exactly and this instead should be `.git | vendor | node_modules`?

Reply:
Yeah, I think we should unify it and also ignore `.git` directory

Reply:
Right now this ignores the.git directory. So this should instead only ignore the git directory and not all hidden folders?

Reply:
Oh, sorry. I didn't read the code fully. I think that still makes sense - what if you try to import from `.dist/` or `.<framework_name>/`? I can see a scenario where people try to do that.

Reply:
With the current code, they would just have to be specific or import a sub folder in that case.

Reply:
Fair enough

Reply:
Actually, they could just do:

```ts
import(`./dist/${expr}`)
```

Thinking about it more, I think we should ignore all hidden folders here because there's this way to import and in many cases I think people wouldn't want it searching hidden folders.